src/integration-test/java/com/commercetools/sync/integration/externalsource/categories/CategorySyncIT.java

```diff
@@ -351,6 +357,24 @@ void syncDrafts_WithNewCategoryWithExistingParent_ShouldCreateCategory() {
         assertThat(syncStatistics).hasValues(1, 1, 0, 0, 0);
     }
 
+    @Test
+    void syncDrafts_WithNewCategoryWithExistingParentWithKey_ShouldNotCreateCategory() {
+        // Category draft coming from external source.
+        final CategoryDraft categoryDraft = CategoryDraftBuilder
+            .of(LocalizedString.of(Locale.ENGLISH, "Modern Furniture"),
+                LocalizedString.of(Locale.ENGLISH, "modern-furniture"))
+            .key("newCategory")
+            .parent(ResourceIdentifier.ofId(oldCategoryKey))
```

Comment:
this test is confusing, why you are adding parent key into id field ?

Reply:
8b42fb59